src/server/goodid/utils.js

```diff
@@ -0,0 +1,80 @@
+import axios from 'axios'
+import { PhoneNumberUtil } from 'google-libphonenumber'
+
+import { substituteParams } from '../utils/axios'
+import { get, toUpper } from 'lodash'
+import { getAgent, getSubjectId } from './veramo'
+
+export class GoodIDUtils {
+  constructor(httpApi, phoneNumberApi, getVeramoAgent) {
+    const http = httpApi()
+    const { request, response } = http.interceptors
+
+    request.use(req => substituteParams(req))
+    response.use(({ data }) => data)
+
+    this.http = http
+    this.phoneUtil = phoneNumberApi.getInstance()
+    this.getVeramoAgent = getVeramoAgent
+  }
+
+  async getCountryCodeFromIPAddress(ip) {
+    const countryCode = await this.http
+      .get('https://get.geojs.io/v1/ip/country/:ip.json', { ip })
+      .then(response => get(response, 'country_3'))
+
+    if (!countryCode) {
+      throw new Error(`Failed to get country code from IP address '${ip}': response is empty or invalid`)
+    }
+
+    return countryCode
+  }
+
+  async getCountryCodeFromGeoLocation(latitude, longitude) {
+    const countryCode = await this.http
+      .get('https://nominatim.openstreetmap.org/reverse', {
+        format: 'jsonv2',
+        lat: latitude,
+        lon: longitude
+      })
+      .then(response => get(response, 'address.country_code'))
+      .then(toUpper)
+
+    if (!countryCode) {
+      throw new Error(
+        `Failed to get country code from coordinates '${latitude}:${longitude}': response is empty or invalid`
+      )
+    }
+
+    return countryCode
+  }
+
+  getCountryCodeFromMobile(phoneNumber) {
+    const { phoneUtil } = this
+    const number = phoneUtil.parseAndKeepRawInput(phoneNumber, 'US')
```

Comment:
whats 'US' for? that doesnt seem like the right method

Reply:
that's default country for the case if number does not contains region code

Reply:
https://github.com/ruimarinho/google-libphonenumber/tree/master?tab=readme-ov-file#methods

```
parseAndKeepRawInput(numberToParse, defaultRegion) - parses a string and returns it in proto buffer format while keeping the raw input value.
```